### PR TITLE
Refactoring of communication with PGP frameworks

### PR DESCRIPTION
**Pull request by SimplyDanny**
Like @ZortacDev already started in #289 this PR removes the `Gopenpgpwrapper` glue code and uses Go's Crypto framework directly to interface with GopenGPG. It incorporates the following sub-steps:
* Removes `Gopenpgpwrapper` glue code.
* Builds `Crypto.framework` using Gomobile.
* Implements workaround for `Data` objects passed to Go code.
* Refactors `PGPAgent` and extract GopenPGP and ObjectivePGP specific code conforming to the same protocol.
* Moves key import from `PGPAgent` to `KeyFileManager`.
* Adds basic tests for the `Crypto.framework` functionality.
* Adapts `PGPAgentTest` to new `PGPAgent` API.

Tests which are supposed to work later on with fixes in 3rd party libraries are commented out. In particular, RSA and EdDSA keys work with GopenPGP as long as they have no sub-keys. RSA with sub-keys works with ObjectivePGP. There is no support for EdDSA with sub-keys yet.

Errors are passed on to the user if reasonable. There are still some improvements possible, though. The `AppError` enumeration could just be more specific in many cases where it is used.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ before_script:
       travis_wait 45 carthage bootstrap --platform iOS
     fi
   - |
-    if [ ! $(git diff --name-only "$TRAVIS_COMMIT_RANGE" | grep -E "gopenpgp_build.sh|gopenpgpwrapper.go") ] && [ -d "go/dist/Gopenpgpwrapper.framework" ]; then
+    if [ ! $(git diff --name-only "$TRAVIS_COMMIT_RANGE" | grep -E "gopenpgp_build.sh|gopenpgpwrapper.go") ] && [ -d "go/dist/Crypto.framework" ]; then
```
Comment (mssun) [inline .travis.yml]: `gopenpgpwrapper.go` can be removed.

**Comment by mssun** [comment]
Thank you so much. Looks good to me.

Does it mean #289 can be closed and we can continue the #293 patch?

**Comment by SimplyDanny** [comment]
> Does it mean #289 can be closed and we can continue the #293 patch?

Yes, as I cannot push to the other branch I decided to open a new PR.

**Comment by mssun** [comment]
Merged, thanks.

**Comment by ZortacDev** [comment]
@SimplyDanny I just took a look at your code changes. Am I correct in the assumption that you expect keys to be in ASCII armored format and don't differentiate between binary and armored messages/password files? Or am I missing something?

**Comment by SimplyDanny** [comment]
Yes, up to now Pass for iOS tells the user in all key import dialogs and in the wiki to export their keys in ASCII-armored format. That's why in this refactoring I did it the simple way and only considered ASCII-armored keys.

**Pull request merged**